src/main/java/com/adwicorp/aanandamsn/service/UserService.java

```diff
@@ -0,0 +1,113 @@
+        try {
+            List<User> userList = userRepository.findByIsDeletedFalse();
+            List<UserResponse> userResponses = userList.stream()
+                    .map(x -> userMapper.mapToUserResponse(x))
+                    .collect(Collectors.toList());
+            return userResponses;
+        } catch (Exception e) {
+            logger.info("UserService::SaveUser::TraceID {}::Error {}", traceId, e.getStackTrace());
+            throw new BusinessException(ErrorCodes.INTERNAL_SERVER_ERROR,
+                    ErrorCodes.ERROR_CODE_MESSAGE_MAP.get(ErrorCodes.INTERNAL_SERVER_ERROR));
+        }
+    }
+
+    public String saveUser(UserRequest userRequest) {
+        String traceId = TraceContext.getTraceId();
+        logger.info("UserService::SaveUser::TraceID {}", traceId);
+        User user = userMapper.mapToUser(userRequest);
+        try {
+            userRepository.save(user);
+            return "User saved successfully";
+        } catch (DataIntegrityViolationException e) {
+            String column = "";
+            Throwable cause = e.getCause().getCause();
+            if (cause != null && cause.getMessage() != null) {
+                String message = cause.getMessage();
+                if (message.contains("entry")) {
+                    int startIndex = message.indexOf("entry") + 7;
+                    int endIndex = message.indexOf(" ", startIndex);
+                    column = message.substring(startIndex, endIndex).trim();
+                }
+            }
+            logger.info("UserService::SaveUser::TraceID {}::Error {}", traceId, e.getStackTrace());
+            throw new BusinessException(ErrorCodes.DB_EXISTING_DATA,
+                    ErrorCodes.ERROR_CODE_MESSAGE_MAP.get(ErrorCodes.DB_EXISTING_DATA) + column);
+        } catch (Exception e) {
+            logger.info("UserService::SaveUser::TraceID {}::Error {}", traceId, e.getStackTrace());
+            throw new BusinessException(ErrorCodes.INTERNAL_SERVER_ERROR,
+                    ErrorCodes.ERROR_CODE_MESSAGE_MAP.get(ErrorCodes.INTERNAL_SERVER_ERROR));
+        }
+
+
+    }
+
+    public String updateUser(Long userId, UserUpdateRequest userUpdateRequest) throws DataIntegrityViolationException{
+        String traceId = TraceContext.getTraceId();
+        logger.info("UserService::UpdateUser::TraceID {}", traceId);
+        try {
+            User user = userRepository.findByUserId(userId);
+            user = userMapper.mapToUserUpdate(user, userUpdateRequest);
+            userRepository.save(user);
+            return "User updated successfully";
+        } catch (Exception e) {
+            logger.info("UserService::SaveUser::TraceID {}::Error {}", traceId, e.getStackTrace());
+            throw new BusinessException(ErrorCodes.INTERNAL_SERVER_ERROR,
+                    ErrorCodes.ERROR_CODE_MESSAGE_MAP.get(ErrorCodes.INTERNAL_SERVER_ERROR));
+        }
+    }
+
+    public String deleteUser(Long userId) {
```

Comment:
Merge with update user. check if modified date time is also updated or not.

Reply:
Done.